tests/coverage/test_processor.py

```
@@ -0,0 +1,436 @@
+        assert isinstance(processor, LcovProcessor), "Expected LcovProcessor instance"
+
+    def test_create_processor_unsupported_type(self):
+        with pytest.raises(
+            ValueError, match="Invalid coverage type specified: unsupported_type"
+        ):
+            CoverageProcessorFactory.create_processor(
+                "unsupported_type", "fake_path", "app.py"
+            )
+
+
+class TestCoverageProcessor:
```

Ruff (0.8.2)

Redefinition of unused `TestCoverageProcessor` from line 50

(F811)